x-pack/plugins/canvas/public/components/sidebar_header/sidebar_header.tsx

```diff
@@ -0,0 +1,329 @@
+    // TODO: restore when group and ungroup can be triggered outside of workpad_page
+    // groupIsSelected: PropTypes.bool,
+    // groupNodes: PropTypes.func.isRequired,
+    // ungroupNodes: PropTypes.func.isRequired,
+  };
+
+  public static defaultProps = {
+    // TODO: restore when group and ungroup can be triggered outside of workpad_page
+    // groupIsSelected: false,
+    showLayerControls: true,
+  };
+
+  public state = {
+    isModalVisible: false,
+  };
+
+  private _isMounted = false;
+  private _showModal = () => this._isMounted && this.setState({ isModalVisible: true });
+  private _hideModal = () => this._isMounted && this.setState({ isModalVisible: false });
+
+  public componentDidMount() {
+    this._isMounted = true;
+  }
+
+  public componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  private _renderLayoutControls = () => {
+    const { bringToFront, bringForward, sendBackward, sendToBack } = this.props;
+    return (
+      <Fragment>
+        <EuiFlexItem grow={false}>
+          <EuiToolTip position="bottom" content="Move element to top layer">
+            <EuiButtonIcon
+              color="text"
+              iconType="sortUp"
+              onClick={bringToFront}
+              aria-label="Move element to top layer"
+            />
+          </EuiToolTip>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiToolTip position="bottom" content="Move element up one layer">
+            <EuiButtonIcon
+              color="text"
+              iconType="arrowUp"
+              onClick={bringForward}
+              aria-label="Move element up one layer"
+            />
+          </EuiToolTip>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiToolTip position="bottom" content="Move element down one layer">
+            <EuiButtonIcon
+              color="text"
+              iconType="arrowDown"
+              onClick={sendBackward}
+              aria-label="Move element down one layer"
+            />
+          </EuiToolTip>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiToolTip position="bottom" content="Move element to bottom layer">
+            <EuiButtonIcon
+              color="text"
+              iconType="sortDown"
+              onClick={sendToBack}
+              aria-label="Move element to bottom layer"
+            />
+          </EuiToolTip>
+        </EuiFlexItem>
+      </Fragment>
+    );
+  };
+
+  private _getLayerMenuItems = () => {
+    const { bringToFront, bringForward, sendBackward, sendToBack } = this.props;
+
+    return {
+      name: 'Order',
+      className: topBorderClassName,
+      panel: {
+        id: 1,
+        title: 'Order',
+        items: [
+          {
+            name: 'Bring to front', // TODO: check against current element position and disable if already top layer
+            icon: 'sortUp',
+            onClick: () => bringToFront(),
```

Comment:
Couldn't this be `onClick: bringToFront`?  Depending on how this collection is used, you may end up forcing a re-render, since this function is newly created every time the function is called.

Reply:
Yes, it could. I refactored the `onClick` handlers in this component.